src/lib/pubkey/rsa/rsa.cpp

```diff
@@ -445,6 +445,20 @@ class RSA_Private_Operation {
             m_max_d1_bits(m_private->p_bits() + m_blinding_bits),
             m_max_d2_bits(m_private->q_bits() + m_blinding_bits) {}
 
+      void raw_op(std::span<const uint8_t> input, std::span<uint8_t> out) {
+         if(input.size() > public_modulus_bytes()) {
+            throw Decoding_Error("RSA input is too long for this key");
+         }
+         const BigInt input_bn(input.data(), input.size());
+         if(input_bn >= m_public->get_n()) {
+            throw Decoding_Error("RSA input is too large for this key");
+         }
+         const BigInt recovered = m_blinder.unblind(rsa_private_op(m_blinder.blind(input_bn)));
+         BOTAN_ASSERT(input_bn == m_public->public_op(recovered), "RSA consistency check");
+         BOTAN_ASSERT(m_public->public_modulus_bytes() == out.size(), "output size check");
+         BigInt::encode_1363(out, recovered);
+      }
+
       secure_vector<uint8_t> raw_op(const uint8_t input[], size_t input_len) {
```

Comment:
Please avoid the duplicated implementation by using the out-param based method. Along those lines:

```C++
secure_vector<uint8_t> raw_op(const uint8_t input[], size_t input_len) {
   secure_vector<uint8_t> out(m_public->public_modulus_bytes);
   raw_op(out, {input, input_len});
   return out;
}
```

... as a follow-up we should probably see if we can get rid of this overload entirely.